torch_xla/distributed/xla_backend.py

```diff
@@ -82,15 +82,33 @@ def allreduce(self, tensors, all_reduce_options):
     return _ret_work(tensors)
 
   # method for dist.all_gather_into_tensor under eager mode.
-  def _allgather_base(self, output_tensor, input_tensor, opts):
-    return self.allgather(output_tensor, input_tensor, opts)
+  def _allgather_base(self, output_tensor: torch.Tensor,
+                      input_tensor: torch.Tensor, opts: AllgatherOptions):
+    is_scalar = (input_tensor.dim() == 0)
```

Comment:
Do you think it'll help if we point to pytorch documentation for all_gather_into_tensor here?